x-pack/plugins/siem/common/endpoint/generate_data.ts

```diff
@@ -75,21 +75,98 @@ const POLICIES: Array<{ name: string; id: string }> = [
 const FILE_OPERATIONS: string[] = ['creation', 'open', 'rename', 'execution', 'deletion'];
 
 interface EventInfo {
-  category: string;
+  category: string | string[];
   /**
    * This denotes the `event.type` field for when an event is created, this can be `start` or `creation`
    */
   creationType: string;
 }
 
+/**
+ * The valid ecs categories.
+ */
+export enum ECSCategory {
```

Comment:
:grey_question: Some stuff I read on Typescript a while back recommended not using `enum` unless it's `const enum` . I don't recall the reason, exactly.

Reply:
Oh good call. I'll make them `export const enum`

Reply:
hmm doesn't seem to work:

```
plugins/siem/common/endpoint/generate_data.test.ts
  ● Test suite failed to run

    SyntaxError: /Users/jbuttner/proj/es/kibana/x-pack/plugins/siem/common/endpoint/generate_data.ts: 'const' enums are not supported.
      86 |  * The valid ecs categories.
      87 |  */
    > 88 | export const enum ECSCategory {
         |        ^
      89 |   Driver = 'driver',
      90 |   File = 'file',
      91 |   Network = 'network',

      at File.buildCodeFrameError (../node_modules/@babel/core/lib/transformation/file/file.js:248:12)
      at NodePath.buildCodeFrameError (../node_modules/@babel/traverse/lib/path/index.js:144:21)
      at transpileEnum (../node_modules/@babel/plugin-transform-typescript/lib/enum.js:25:16)
      at PluginPass.TSEnumDeclaration (../node_modules/@babel/plugin-transform-typescript/lib/index.js:327:27)
      at newFn (../node_modules/@babel/traverse/lib/visitors.js:179:21)
      at NodePath._call (../node_modules/@babel/traverse/lib/path/context.js:55:20)
      at NodePath.call (../node_modules/@babel/traverse/lib/path/context.js:42:17)
      at NodePath.visit (../node_modules/@babel/traverse/lib/path/context.js:90:31)
      at TraversalContext.visitQueue (../node_modules/@babel/traverse/lib/context.js:112:16)
      at TraversalContext.visitSingle (../node_modules/@babel/traverse/lib/context.js:84:19)
      at TraversalContext.visit (../node_modules/@babel/traverse/lib/context.js:140:19)
      at Function.traverse.node (../node_modules/@babel/traverse/lib/index.js:84:17)
      at NodePath.visit (../node_modules/@babel/traverse/lib/path/context.js:97:18)
      at TraversalContext.visitQueue (../node_modules/@babel/traverse/lib/context.js:112:16)
      at TraversalContext.visitMultiple (../node_modules/@babel/traverse/lib/context.js:79:17)
      at TraversalContext.visit (../node_modules/@babel/traverse/lib/context.js:138:19)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        3.42s
```

Reply:
Oh I see... I never used them, but I do remember Robert saying something about how the way we have our build set up won't allow us to use those - so carry on.